src/frontends/tensorflow/src/op/while.cpp

```diff
@@ -42,9 +42,9 @@ OutputVector translate_while_op(const NodeContext& node) {
         body_model,
         "[TensorFlow Frontend] Internal error or incorrect input model. Cannot find body graph with name " + body_type);
 
-    auto loop = create_loop_for_tf_while(node.get_name(), body_model, cond_model, ov_inputs);
-    set_node_name(node.get_name(), loop);
-    return loop->outputs();
+    auto while_loop = create_loop_for_tf_while(node.get_name(), body_model, cond_model, ov_inputs);
+    set_node_name(node.get_name(), while_loop.second);
```

Comment:
please do not use `set_node_name` (it will incorrectly count ports due to skip connections) and just set tensor names